File: board-room/src/main/java/net/stzups/board/room/server/websocket/protocol/server/messages/ServerMessageHandshake.java
package net.stzups.board.room.server.websocket.protocol.server.messages;

import io.netty.buffer.ByteBuf;
import net.stzups.board.data.objects.session.PersistentHttpSession;
import net.stzups.board.room.server.websocket.protocol.server.ServerMessage;
import net.stzups.board.room.server.websocket.protocol.server.ServerMessageType;

public class ServerMessageHandshake extends ServerMessage {
    private long id;
    private long token;

    public ServerMessageHandshake(PersistentHttpSession persistentHttpSession) {
        super(ServerMessageType.HANDSHAKE);
        this.id = persistentHttpSession.getId();
        //this.token = persistentHttpSession.generateToken();
    }

    @Override
    public void serialize(ByteBuf bytebuf) {
        super.serialize(bytebuf);
        bytebuf.writeLong(id);
        bytebuf.writeLong(token);
    }
}

File: scribbleshare-app/scribbleshare-frontend/src/scripts/common/protocol/server/messages/ServerMessageHandshake.ts
import ServerMessage from "../ServerMessage.js";
import ServerMessageType from "../ServerMessageType.js";
import ByteBuffer from "../../ByteBuffer.js";

export default class ServerMessageHandshake extends ServerMessage {
    client: number;

    constructor(byteBuffer: ByteBuffer) {
        super();
        this.client = byteBuffer.readInt16();
    }

    getType(): ServerMessageType {
        return ServerMessageType.HANDSHAKE;
    }
}